src/NSrtm.Core/Pgm/DataDesciption/PgmDataDescriptionExtractor.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace NSrtm.Core.Pgm
+{
+    internal static class PgmPreamble
+    {
+        public static readonly IReadOnlyDictionary<string, Regex> Extractor =
+            new Dictionary<string, Regex>
+            {
+                {"MagicNumber", new Regex(@"^(P5)[^$]", RegexOptions.Multiline)},
+                {"Offset", new Regex(@"^#[\s]Offset[\s]+(?<offset>.+)$", RegexOptions.Multiline)},
+                {"Scale", new Regex(@"^#[\s]Scale[\s]+(?<scale>.+)$", RegexOptions.Multiline)},
+                {"Origin", new Regex(@"^#[\s]Origin[\s](?<lat>\d+)\D+(?<lon>\d+)", RegexOptions.Multiline)},
+                {"PointParameters", new Regex(@"(\n|\r|\r\n)(?<width>\d+)\s+(?<height>\d+)(\n|\r|\r\n)(?<maxValue>\d+)", RegexOptions.Multiline)},
+            };
+    }
+
+    internal static class PgmDataDescriptionExtractor
+    {
+        public static PgmDataDescription FromStream(Stream stream)
+        {
+            var preamble = extractPreambleFromStream(stream);
+            return getConstatantsFromPreamble(preamble);
+        }
+
+        private static string extractPreambleFromStream(Stream stream)
+        {
+            using (var reader = new BinaryReader(stream, Encoding.UTF8, true))
```

Comment:
Because i want to use BinaryReader(Stream, Encoding, Boolean) constructor and this encoding is default encoding for BinaryReader(Stream)